torch/csrc/utils/future.h

```diff
@@ -156,6 +153,24 @@ class TORCH_API Future final {
     }
   }
 
+  void markCompletedInternal(T value,
+      std::unique_lock<std::mutex>& lock) {
```

Comment:
It takes a reference, while you are passing rvalue with `std::move`. 

Do you want to move a rvalue instead of passing reference?

`std::unique_lock` should be movable.

Change to `std::unique_lock<std::mutex> lock`?